CMakeLists.txt

```diff
@@ -110,6 +112,38 @@ option(REQUIRE_PTHREADS "For non-Windows systems, set whether Csound will use th
 option(BUILD_DOCS "Build documentation (requires doxygen)" OFF)
 option(DEBUG_ERROR_ON_WARNING "For a debug build, error on a compiler# warning" OFF)
 option(BARE_METAL "To configure for bare metal build" OFF)
+option(ADD_COMMANDLINE_TESTS "Add command-line tests" ON)
+option(TRACK_COVERAGE "Track coverage" OFF)
+
+if (BUILD_TESTS)
+    # c tests require the static library
+    set(BUILD_STATIC_LIBRARY ON)
+endif()
+
+if (TRACK_COVERAGE)
+    check_cxx_compiler_flag(-g has_debug_info)
+    if (has_debug_optimization)
+        # disable optimizations for better coverage stats
+        add_compile_options(-Og)
```

Comment:
I'm not sure I like this. Someone enabling coverage would not expect all of a sudden that the binary output would be extremely slow. I think the coverage build should be a separate thing that is built separately in the CI process rather potentially affecting the main binary output.

Reply:
> coverage build should be a separate target that is built separately in the CI process

Hmm, what do you mean?

Reply:
Build a separate CMake target just for the coverage so there is no chance of contaminating the main build. It probably should exclude the opcodes actually and just test the engine. Also you can specify what compile options like no optimisation. It should probably be just a debug build I guess.

Reply:
Removed coverage, separate PR to follow